lib/TPP/MapConvToMatmul.cpp

```diff
@@ -276,9 +276,15 @@ mlir::linalgx::mapConvToMatmul(RewriterBase &rewriter,
   };
 
   Location loc = linalgOp.getLoc();
-  linalg::GenerateLoopNest<scf::ForOp>::doit(
-      rewriter, loc, loopRanges, linalgOp, linalgOp.getIteratorTypesArray(),
-      gemmBuilder);
+  if (linalgOp.hasBufferSemantics()) {
+    linalg::GenerateLoopNest<scf::ParallelOp>::doit(
```

Comment:
Does this mean that linalg with memrefs is guaranteed to not have dependences resolved through memory?